### PR TITLE
remote/client: add write-image-like write-file command

### PR DESCRIPTION
**Pull request by a3f**
**Description**

- what do you use the feature for?

Interactive use with `labgrid-client`. It's useful for bootloader development for processors with bootroms that load code from file systems (e.g. EFI system partition, OMAP, AT91, Zynq, Raspberry ... etc.)

- how does labgrid benefit as a testing library from the feature?

I guess a unit test could place e.g. a broken configuration file and see how it behaves?

- how did you verify the feature works?

Remote usbsdmux that I usually use with `write-image`. Tried writing non-existing files and overwriting existing files with `write-file`. Works well.

- if hardware is needed for the feature, which hardware is supported and which
  hardware did you test with?

Anything exposing a `USBStorageDriver`. I tested with a usbsdmux.

**Checklist**
- [x] Documentation for the feature
- [-] Tests for the feature 
- [x] The arguments and description in doc/configuration.rst have been updated
- [-] Add a section on how to use the feature to doc/usage.rst
- [-] Add a section on how to use the feature to doc/development.rst
- [x] CHANGES.rst has been updated
- [x] PR has been tested
- [x] Man pages have been regenerated

### CODE REVIEW

**Comment by Emantor** [comment]
#548

File: debian/control [hunk 1]
```diff
@@ -10,7 +10,7 @@ Package: labgrid
 Architecture: any
 Pre-Depends: dpkg (>= 1.16.1), python3, ${misc:Pre-Depends}
 Depends: ${python3:Depends}, ${misc:Depends}, ${shlibs:Depends}
-Recommends: openssh-client, microcom, socat, sshfs, rsync
+Recommends: openssh-client, microcom, socat, sshfs, rsync, pmount
```
Comment (jluebbe) [inline debian/control:13]: Why do you prefer `pmount` over `udiskctl`?

File: labgrid/driver/usbstoragedriver.py [hunk 2]
```diff
@@ -152,6 +160,67 @@ def get_size(self):
...
+        self.await_medium()
+
+        partition = "" if partition is None else partition
+        remote_path = mf.get_remote_path()
+        target = "{}{}".format(self.storage.path, partition)
+
+        label = escape(target)
+        dstpath = os.path.normpath(dstpath)
+        if not dstpath.startswith("/"):
+            raise ValueError("dstpath must be ab absolute path")
```
Comment (jluebbe) [inline labgrid/driver/usbstoragedriver.py:189]:
```suggestion
            raise ValueError("dstpath must be an absolute path")
```

File: labgrid/driver/usbstoragedriver.py [hunk 3]
```diff
@@ -152,6 +160,67 @@ def get_size(self):
...
+
+        dstpath = "/media/{}/{}".format(label, dstpath)
```
Comment (jluebbe) [inline labgrid/driver/usbstoragedriver.py:191]: Why can we expect `/media/{label}/` to be writable? Shouldn't the mount helper tell us where the filesystem was mounted?

File: labgrid/driver/usbstoragedriver.py [hunk 4]
```diff
@@ -152,6 +160,67 @@ def get_size(self):
...
+
+        args = ["pumount {}".format(target)]
+
+        processwrapper.check_output(
+            self.storage.command_prefix + args,
+            print_on_silent_log=True
+        )
+
+        if caught:
+            raise caught
```
Comment (jluebbe) [inline labgrid/driver/usbstoragedriver.py:222]: Please use a more idiomatic try:/finally: structure.

**Review by jluebbe — CHANGES_REQUESTED** [review]

**Comment by Emantor** [comment]
Closing this, please add udisks support instead of pmount and open a new PR.

File: labgrid/driver/usbstoragedriver.py (hunk 4 above)
Comment (a3f) [reply labgrid/driver/usbstoragedriver.py:222]: I might've never used finally before, so I had forgotten about it. Thanks! :)

File: debian/control (hunk 1 above)
Comment (a3f) [reply debian/control:13]: Will resubmit with udisksctl

File: labgrid/driver/usbstoragedriver.py (hunk 3 above)
Comment (a3f) [reply labgrid/driver/usbstoragedriver.py:191]: udisksctl does, albeit a bit more verbose than necessary. Will parse it.

**Pull request closed without merging**